konrad/cloud.py

```diff
@@ -263,6 +278,51 @@ def __init__(self, z, cloud_fraction, mass_water, mass_ice,
             droplet_radius=droplet_radius
         )
 
+    def __add__(self, other):
+        """Define the superposition of two clouds in a layer."""
+        if not np.allclose(self.z, other.z):
+            raise ValueError('Incompatible `z` grids.')
+
+        cldf = np.max(
+            (self.cloud_area_fraction_in_atmosphere_layer,
+             other.cloud_area_fraction_in_atmosphere_layer),
+            axis=0,
+        ).clip(min=0, max=1)
+
+        iwc = np.max(
+            (self.mass_content_of_cloud_ice_in_atmosphere_layer,
+             other.mass_content_of_cloud_ice_in_atmosphere_layer),
+            axis=0,
+        )
+
+        lwc = np.max(
+            (self.mass_content_of_cloud_liquid_water_in_atmosphere_layer,
+             other.mass_content_of_cloud_liquid_water_in_atmosphere_layer),
+            axis=0,
+        )
+
+        ips = np.mean(
+            (self.cloud_ice_particle_size, other.cloud_ice_particle_size),
+            axis=0,
+        )
+
+        dr = np.mean(
+            (self.cloud_water_droplet_radius,
```

Comment:
To be consistent with the way the direct input clouds are added, the cloud ice and droplet sizes should be ones that maximise the optical depth, which is normally the smallest value.

Reply:
But okay for now this is not important, because we will avoid having two clouds in the same model layer.

Reply:
I can change it anyhow 😉 Using a reasonable implementation now reduces the risk of unexpected behaviour in the future, when we forgot that our clouds should not overlap

Reply:
Should we use `np.sum` for masses and `np.min` for sizes? Or is this more extreme than our assumption of cumulative optical thickness?

Reply:
I wanted to say take `np.max` for masses, `np.min` for sizes and for optical depth (in the other cloud class) `np.max`, to have a biggest / strongest cloud wins policy! But, I just realised this doesn't really work, because the cloud with the bigger mass, might also have a larger particle size... As a simple approach (easily made consistent between `DirectInputCloud` and `PhysicalCloud`), we could take the cloud with the largest area fraction as the winning cloud, ignore any other clouds in that layer, and take all the cloud properties from the winning cloud. What do you think?